### ISSUE TITLE
[feature] A directive to insert / stamp out TemplateRefs

### ISSUE THREAD

**Issue by pkozlowski-opensource** [angular/angular#7615]
There are cases where we've got a reference to a `TemplateRef` and would like to use it to stamp out live HTML fragments. Currently it can be easily done from the JS side (`viewContainerRef.createEmbeddedView(tplRef)`) but this issue is about creating a directive that would allow us to do this "inserting" from a template.

A "workish" example of such directive:

```javascript
@Directive({
    selector: 'ng-insert-tpl'
})
export class NgInsertTpl {
    constructor(private _viewContainerRef: ViewContainerRef) {
    }
    
    @Input() set template(tplRef: TemplateRef) {
        this._viewContainerRef.createEmbeddedView(tplRef);
    }
}
```

Naming things is hard so `ng-insert-tpl` is definitively up to debate.

This is part of https://github.com/angular/angular/issues/7546#issuecomment-196848167, it could be used in action like so: http://plnkr.co/edit/sleVVrgqo66JN9G48yyj?p=preview

**Comment by jelbourn** [comment]
@pkozlowski-opensource we've got something like this already in material. 
https://github.com/angular/material2/tree/master/src/core/portal

It's slightly different, because you can use it to attach either a `TemplateRef` _or_ a component instance (with `DynamicComponentLoader`).

**Comment by pkozlowski-opensource** [comment]
@jelbourn right, so this just confirms that we are converging in terms of needs and IMO this means that there is an abstraction here that would make sense in the core. I've discussed this today with Tobias and our thinking was to add a directive to a core.

Comparing portal to the proposal in this issue I can see 2 differences:
* ability to pass a component or a template ref -> makes sense, this would be actually beneficial for cases like tooltips / popovers for me
* ability to pass local vars -> definitively makes sense, was discussing this with Tobias but wanted to keep it simple. But yeh, would love to have local vars as well.

In short: I think that we both see a need here and I would love that the identified mechanism end up in core as I can see it being used in many cases (not only for widget libraries).

**Comment by mhevery** [comment]
+1 We should do this in core.
